lib/new.js

```diff
@@ -0,0 +1,687 @@
+/**
+ * Copyright 2018 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const path = require('path');
+const puppeteer = require('puppeteer-core');const findChrome = require('./find_chrome');
+const {rpc} = require('../rpc');
+const debugApp = require('debug')('carlo:app');
+const debugServer = require('debug')('carlo:server');
+const {Color} = require('./color');
+const {HttpRequest} = require('./http_request');
+
+const fs = require('fs');
+const util = require('util');
+const {URL} = require('url');
+const EventEmitter = require('events');
+const fsReadFile = util.promisify(fs.readFile);
+
+let testMode = false;
+
+class App extends EventEmitter {
+  /**
+   * @param {!Puppeteer.Browser} browser Puppeteer browser
+   * @param {!Object} options
+   */
+  constructor(browser, options) {
+    super();
+    this.browser_ = browser;
+    this.options_ = options;    this.windows_ = new Map();    this.exposedFunctions_ = [];
+    this.pendingWindows_ = new Map();
+    this.windowSeq_ = 0;
```

Comment:
format: style mismatch:
```javascript
44|    this.pendingWindows_ = new Map();
45|    this.windowSeq_ = 0;
46|    this.www_ = [];
```
```suggestion
    this.windowSeq_ = 0;this.www_ = [];}
```

⏎ at column 25 should be removed.
Triggered rule # 253
```
  +1.roles not in {ARGUMENT, CALLEE, LITERAL}
	∧ +1.reserved not in {=, {}
	∧ +2.length ≤ 49
	∧ +2.roles not in {RIGHT}
	∧ +3.roles not in {BLOCK}
	∧ ^1.roles not in {IDENTIFIER, OPERATOR, DECLARATION}
	∧ ^1.internal_type not in {BlockStatement}
	∧ -1.roles not in {STRING, EXPRESSION}
	∧ -1.reserved not in {,, :, ), .}
	∧ -1.diff_offset ≤ 14
	∧ -2.diff_line = 0
	∧ -2.roles in {CALL}
	∧ -2.reserved not in {(}
	∧ -2.label not in {<-space>}
	∧ -2.diff_col ≥ 4
	∧ -3.label not in {<newline>}
	∧ -4.roles not in {CALL}
	∧ -4.reserved not in {;}
	∧ -5.label not in {<newline>}
	⇒ y = ∅
	Confidence: 0.997. Support: 145.
```
⏎␣⁻␣⁻ at column 20 should be removed.
Triggered rule # 182
```
  +1.roles not in {ARGUMENT}
	∧ +1.internal_type not in {StringLiteral}
	∧ +2.length ≤ 6
	∧ +2.roles not in {RIGHT}
	∧ +3.roles not in {NUMBER}
	∧ +4.reserved not in {,}
	∧ +5.reserved not in {(}
	∧ ^1.roles in {OR, OPERATOR} and not in {DECLARATION}
	∧ -1.roles not in {STRING, EXPRESSION}
	∧ -1.reserved not in {,, :, (, .}
	∧ -2.diff_line = 0
	∧ -4.reserved not in {;}
	∧ -4.diff_offset ≤ 6
	⇒ y = ␣
	Confidence: 0.938. Support: 8.
```